# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../iptraf/compile-fix.patch
# Copyright (C) 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur iptraf-3.0.0-orig/src/othptab.c iptraf-3.0.0/src/othptab.c
--- iptraf-3.0.0-orig/src/othptab.c	2005-09-13 06:42:54.000000000 +0000
+++ iptraf-3.0.0/src/othptab.c	2005-12-28 19:01:08.000000000 +0000
@@ -17,13 +17,14 @@
 ***/
 
 #include <asm/types.h>
-#include <linux/if_ether.h>
-#include <linux/if_tr.h>
-#include <linux/if_fddi.h>
+#include "tcptable.h"
+//#include <linux/if_ether.h>
+//#include <linux/if_tr.h>
+//#include <linux/if_fddi.h>
 #include <winops.h>
 #include "arphdr.h"
 #include "options.h"
-#include "tcptable.h"
+//#include "tcptable.h"
 #include "othptab.h"
 #include "deskman.h"
 #include "attrs.h"
diff -Nur iptraf-3.0.0-orig/src/packet.c iptraf-3.0.0/src/packet.c
--- iptraf-3.0.0-orig/src/packet.c	2005-09-13 06:42:54.000000000 +0000
+++ iptraf-3.0.0/src/packet.c	2005-12-28 19:02:18.000000000 +0000
@@ -32,6 +32,9 @@
 #include <sys/time.h>
 #include <net/if_arp.h>
 #include <net/if.h>
+#ifdef _NET_IF_H
+#define _LINUX_IF_H
+#endif
 #include <sys/ioctl.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
diff -Nur iptraf-3.0.0-orig/src/tcptable.h iptraf-3.0.0/src/tcptable.h
--- iptraf-3.0.0-orig/src/tcptable.h	2005-09-13 06:42:54.000000000 +0000
+++ iptraf-3.0.0/src/tcptable.h	2005-12-28 18:59:23.000000000 +0000
@@ -20,11 +20,15 @@
 #include <arpa/inet.h>
 #include <net/if_arp.h>
 #include <asm/types.h>
+#include <net/if.h>
+#ifdef _NET_IF_H
+#define _LINUX_IF_H
+#endif
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
 #include <linux/if_tr.h>
-#include <net/if.h>
+//#include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
 #include "servname.h"
diff -Nur iptraf-3.0.0-orig/src/tr.c iptraf-3.0.0/src/tr.c
--- iptraf-3.0.0-orig/src/tr.c	2005-09-13 06:42:54.000000000 +0000
+++ iptraf-3.0.0/src/tr.c	2005-12-28 19:07:49.000000000 +0000
@@ -8,7 +8,7 @@
 
 #include <asm/types.h>
 #include <linux/if_tr.h>
-#include <netinet/in.h>
+//#include <netinet/in.h>
 
 unsigned int get_tr_ip_offset(unsigned char *pkt)
 {
